packages/@expo/cli/src/export/exportAssets.ts

```diff
@@ -92,16 +141,39 @@ export async function exportAssetsAsync(
   );
 
   if (assets[0]?.fileHashes) {
+    debug(`Assets = ${JSON.stringify(assets, null, 2)}`);
+    // Updates the manifest to reflect additional asset bundling + configs
+    // Get only asset strings for assets we will save
+    await resolveAssetPatternsToBeBundledAsync(projectRoot, exp, assets);
+    const bundledAssetsSet = (exp as any).bundledAssets;
+    let filteredAssets = assets;
+    if (bundledAssetsSet) {
+      debug(`Bundled assets = ${JSON.stringify([...bundledAssetsSet], null, 2)}`);
+      // Filter asset objects to only ones that include assetBundlePatterns matches
+      filteredAssets = assets.filter((asset) =>
+        assetShouldBeIncludedInExport(asset, bundledAssetsSet)
+      );
+      debug(`Filtered assets count = ${filteredAssets.length}`);
+    }
     Log.log('Saving assets');
-    await saveAssetsAsync(projectRoot, { assets, outputDir });
+    await saveAssetsAsync(projectRoot, { assets: filteredAssets, outputDir });
   }
 
   // Add google services file if it exists
   await resolveGoogleServicesFile(projectRoot, exp);
 
-  // Updates the manifest to reflect additional asset bundling + configs
-  await resolveAssetBundlePatternsAsync(projectRoot, exp, assets);
-
+  bundles.ios?.assets.forEach((asset: any) => {
```

Comment:
rather than cast to any, prefer to add `embedded?: true` to the typescript definition